shop/src/main/java/ro/msg/learning/shop/controller/LocationController.java

```diff
@@ -0,0 +1,38 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.dto.LocationDto;
+import ro.msg.learning.shop.dto.save.SaveLocationDto;
+import ro.msg.learning.shop.exception.EntityNotFoundException;
+import ro.msg.learning.shop.model.Location;
+import ro.msg.learning.shop.service.LocationService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/locations")
+public class LocationController {
+    private final LocationService locationService;
+
+    public LocationController(LocationService locationService) {
+        this.locationService = locationService;
+    }
+
+    @PostMapping
+    public LocationDto saveLocation(@RequestBody SaveLocationDto location) {
+        System.out.printf("%s\n", location.toString());

```

Comment:
Not generally a good practice to print the exception. However, you could take the desired message and throw the exception :) . This can later be used with an Exception Handler and with Validations. Even exceptions can be clean ;)
https://www.baeldung.com/exception-handling-for-rest-with-spring
https://www.baeldung.com/spring-boot-bean-validation